### PR TITLE
Unit tests that make sure object selector works with CRD

### PR DESCRIPTION
**Pull request by caesarxuchao**
followup of #78505, only the last commit is new.

/sig api-machinery
/assign @liggitt @jpbetz 
/release-note-none

```release-note
NONE
```

### CODE REVIEW

**Comment by liggitt** [comment]
looks good, go ahead and rebase and we'll get this in in v1.16

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go [hunk 1]
```diff
@@ -727,6 +752,37 @@ func NewMutatingTestCases(url *url.URL) []MutatingTest {
...
+				Rules:             matchEverythingRules,
+				NamespaceSelector: &metav1.LabelSelector{},
+				ObjectSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"remove": "me",
+					},
+				},
+				AdmissionReviewVersions: []string{"v1beta1"},
+			}, {
+				Name:              "shouldNotBeCalled",
```
Comment (jpbetz) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go]: Include one test where the selector does match?

**Review by jpbetz — COMMENTED** [review]
Consider adding a test where the selector does match. Looks good otherwise.

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/testing/testcase.go]: the first webhook in this set is called

**Comment by liggitt** [comment]
you might need to rebase on master, or git might be smart about the already merged commits

**Pull request merged**